specviz/widgets/workspace.py

```diff
@@ -512,15 +513,18 @@ def load_data(self, file_path, file_loader=None, display=False):
             return data_items
 
         except:
-            message_box = QMessageBox()
-            message_box.setText("Error loading data set.")
-            message_box.setIcon(QMessageBox.Critical)
-            message_box.setInformativeText(
-                "{}\n{}".format(
-                    sys.exc_info()[0], sys.exc_info()[1])
-            )
-
-            message_box.exec()
+            if _interactive:
```

Comment:
I'm not a _huge_ fan of the private method parameter here. Might it be better to move the qt-specific part of this method into the `_on_load_data` method (to keep those functions together), and reserve this one for the simple loading machinery?

Reply:
@nmearl yes, that makes sense. Some related refactoring is going to be required for my proposed feature in #609 as well.